### PR TITLE
Azure sg2

### PR DESCRIPTION
**Pull request by rguthriemsft**
This pull request contains changes to support delete security group in deck.  Please review so I can submit a pull request to spinnaker main.

### CODE REVIEW

File: app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js [hunk 1]
```diff
@@ -90,20 +99,74 @@ module.exports = angular.module('spinnaker.azure.securityGroup.azure.edit.contro
     }
 
     this.addRule = function(ruleset) {
-      ruleset.push({});
+      ruleset.push({
+        name: $scope.securityGroup.name + '-Rule' + ruleset.length,
+        priority: ruleset.length == 0 ? 100 : 100 * (ruleset.length + 1),
```
Comment (JargoonPard) [inline app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js]: == -> ===

File: app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js [hunk 2]
```diff
@@ -90,20 +99,74 @@ module.exports = angular.module('spinnaker.azure.securityGroup.azure.edit.contro
...
+        ruleset[index].destinationPortRange =
+            ruleset[index].startPort + '-' + ruleset[index].endPort;
     };
 
     this.removeRule = function(ruleset, index) {
       ruleset.splice(index, 1);
     };
 
+    this.moveUp = function(ruleset, index) {
+      if(index == 0)
```
Comment (JargoonPard) [inline app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js]: == -> ===

File: app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js [hunk 3]
```diff
@@ -90,20 +99,74 @@ module.exports = angular.module('spinnaker.azure.securityGroup.azure.edit.contro
...
+        return;
+      swapRules(ruleset, index, index - 1);
+    };
+    this.moveDown = function(ruleset, index) {
+      if(index == ruleset.length - 1)
```
Comment (JargoonPard) [inline app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js]: == -> ===

File: app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js [hunk 4]
```diff
@@ -90,20 +99,74 @@ module.exports = angular.module('spinnaker.azure.securityGroup.azure.edit.contro
...
+      priorityB = ruleset[b].priority;
+      //swap elements
+      ruleset[b] = ruleset[a];
+      ruleset[a] = temp;
+      //swap priorities
+      ruleset[a].priority = priorityA;
+      ruleset[b].priority = priorityB;
+    }
+
+    //$scope.securityGroup.securityRules = [];
```
Comment (JargoonPard) [inline app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js]: remove this line

File: app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js [hunk 5]
```diff
@@ -112,5 +175,5 @@ module.exports = angular.module('spinnaker.azure.securityGroup.azure.edit.contro
       $modalInstance.dismiss();
     };
 
-    initializeSecurityGroups();
+    //initializeSecurityGroups();
   });
```
Comment (JargoonPard) [inline app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js]: remove this line

**Comment by JargoonPard** [comment]
Can you also take a look at the comments that Chris made on the previous PR and make those corrections as well? They were minor style comments that he agreed to let slip and I said we would correct on the next PR.

File: app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js [hunk 6]
```diff
@@ -8,17 +8,24 @@ module.exports = angular.module('spinnaker.azure.securityGroup.azure.edit.contro
...
                                                 taskMonitorService, cacheInitializer, infrastructureCaches,
-                                                _, application, securityGroup, securityGroupWriter) {
+                                                _, application, securityGroup, azureSecurityGroupWriter) {
 
     $scope.pages = {
       ingress: require('./createSecurityGroupIngress.html'),
     };
 
+    securityGroup.securityRules = _.map(securityGroup.securityRules,function(rule){
+      var temp = rule.destinationPortRange.split('-');
```
Comment (JargoonPard) [inline app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js]: Add a space before the {. Netflix is very fussy about syntax.

File: app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js [hunk 7]
```diff
@@ -32,37 +39,39 @@ module.exports = angular.module('spinnaker.azure.securityGroup.azure.edit.contro
...
+//           return {
+//             cidr: rule.range.ip + rule.range.cidr,
+//             type: rule.protocol,
+//             startPort: portRange.startPort,
+//             endPort: portRange.endPort
+//           };
+//         });
+//       })
+//       .flatten()
+//       .value();
```
Comment (JargoonPard) [inline app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js]: You should delete the lines above rather than comment them out.

**Comment by rguthriemsft** [comment]
closing

**Pull request closed without merging**